language/vm/vm-runtime/src/code_cache/module_cache.rs

```diff
@@ -234,14 +239,14 @@ impl<'alloc> VMModuleCache<'alloc> {
                 Ok(Type::Struct(struct_def))
             }
             SignatureToken::Reference(sub_tok) => {
-                let inner_ty =
+                let inner_layout =
```

Comment:
layout is a horrible name for types, it would be good to revert this change and avoid needless churn

Reply:
So just to clarify: I use the name "layout" because this is what the data structure currently is -- it is **not** a nominal type but a description of the data layout of the values of a type. I would even suggest we rename `enum Type` to `Layout` or `TypeLayout` or `DataLayout` or `ValueLayout` to reduce confusion.

On the other hand, if you ask me what `fn resolve_signature_token` should return I will agree it should a "type". More precisely a data structure that contains both the fully qualified type name & data layouts. (Is this the reason you raised the question?) My understanding is that we will eventually change to this model and rewrite the existing resolution/caching logic drastically. Based on this I don't see much value naming variables in a way that "pretends" these are types (which they are not). Nonetheless I really don't have strong opinions on this, and I believe you are likely more informed than me on what the future VM types will look like, so I'll leave the final call to you.

Reply:
I guess I see your point... 
I suppose I was reacting to the change with the following in mind (albeit not articulated at all):
1- in a PR like this and given the changes we are going to make, having "random" renaming just makes the PR heavier with no benefit
2 - as you mentioned the module cache should return types and nothing related to layout strictly speaking

I would appreciate if you could keep changes to a minimum particularly when there are much more significant things to follow up in this PR.
Clean ups can be done at a later time when not exactly in the context of what we are doing here

Reply:
I get your point. I'll remove some name/style changes and try follow the "local" naming conventions to minimize the mental burden on the reviewers. Thanks for pointing this out.